### PR TITLE
Scaling fix from external meshes

### PR DESCRIPTION
**Pull request by AlbertoRemusIIT**
Function `BBFromExternalShape` had an issue since scaling of meshes was not kept into account, in particular:

extract scaling factor from mesh and rescale vertices
```
        double scaling_factor=extMesh->scale.getVal(0);
        for(size_t i=0; i < vertexVector.size(); i++)
        {
            vertexVector[i]*=scaling_factor;
        }
        
```

Rescale **t** vector in transformation **H**

```
        double scaledX = extMesh->link_H_geometry.getPosition().getVal(0)*scaling_factor;
        double scaledY = extMesh->link_H_geometry.getPosition().getVal(1)*scaling_factor;
        double scaledZ = extMesh->link_H_geometry.getPosition().getVal(2)*scaling_factor;
        
        
        Position scaledPosition(scaledX, scaledY, scaledZ);
        
        
        Transform scaled_link_H_geometry(extMesh->link_H_geometry.getRotation(), scaledPosition);
        // Workaround for bug
        Position offset_bb_wrt_link = extMesh->link_H_geometry*offset_bb_wrt_geom;

        box.link_H_geometry = Transform(extMesh->link_H_geometry.getRotation(),
                                         offset_bb_wrt_link);

```

### CODE REVIEW

File: src/model/src/InertialParametersHelpers.cpp [hunk 1]
```diff
@@ -105,6 +109,8 @@ void buildMesh(const aiScene* scene, const aiNode* node, const double scale, std
 
 bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
 {
+    
+
```
Comment (traversaro) [inline src/model/src/InertialParametersHelpers.cpp]: Not a big problem in this case, but in general it is a good idea to avoid to include unrelated changes in a PR, to simplify the reviewers' job.

File: src/model/src/InertialParametersHelpers.cpp [hunk 2]
```diff
@@ -113,24 +119,33 @@ bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
     if (pScene)
     {
         // Extract vector of vertices
-        double scale = 1.0;
+        double scale = 1;
```
Comment (traversaro) [inline src/model/src/InertialParametersHelpers.cpp]: See previous comment.

File: src/model/src/InertialParametersHelpers.cpp [hunk 3]
```diff
@@ -113,24 +119,33 @@ bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
...
             return false;
         }
-
+        
+        
+        double scaling_factor=extMesh->scale.getVal(0);
+        for(size_t i=0; i < vertexVector.size(); i++)
+        {
+            vertexVector[i]*=scaling_factor;
+        }
```
Comment (traversaro) [inline src/model/src/InertialParametersHelpers.cpp]: Even if I am not enthusiastic about it, in the URDF the scale attribute is a three valued parameters (see https://github.com/PR2/pr2_common/blob/6f96993793230c81737478729c4e324c3574424a/pr2_description/urdf/sensors/kinect_camera.urdf.xacro#L25). For this reason, we should not just consider the first value of the scale, but scale every vertex for the different scaling value, even if in this case the scaling vector is the same for all axis.

Comment (traversaro) [reply src/model/src/InertialParametersHelpers.cpp]: See https://answers.ros.org/question/273434/how-to-resize-meshes-in-urdf/ for a related ROS Answer question.

File: src/model/src/InertialParametersHelpers.cpp [hunk 4]
```diff
@@ -172,14 +188,24 @@ bool BBFromExternalShape(ExternalMesh* extMesh, Box& box)
         offset_bb_wrt_geom(0) = (maxX+minX)/2.0;
         offset_bb_wrt_geom(1) = (maxY+minY)/2.0;
         offset_bb_wrt_geom(2) = (maxZ+minZ)/2.0;
-
+        
+        double scaledX = extMesh->link_H_geometry.getPosition().getVal(0)*scaling_factor;
+        double scaledY = extMesh->link_H_geometry.getPosition().getVal(1)*scaling_factor;
+        double scaledZ = extMesh->link_H_geometry.getPosition().getVal(2)*scaling_factor;
```
Comment (traversaro) [inline src/model/src/InertialParametersHelpers.cpp]: As you can read in URDF specs (http://wiki.ros.org/urdf/XML/link) the `scale` attribute is only relevant to the mesh, not to the link_H_geometry transform that is specified via the `visual/origin` tag, and that is already expressed in meters as all the rest of quantities in iDynTree, so there is not need to scale the transform offset.

**Comment by AlbertoRemusIIT** [comment]
changes in my fork of iDynTree
